### PR TITLE
[FS] FederationWalletSyncManager uses a queue

### PR DESCRIPTION
**Pull request by codingupastorm**
Implementation virtually copied from our own WalletSyncManager.

There's no tests for this component but the integration tests (including our skipped one that demonstrates a transfer) are running (and a little bit faster, too).

### CODE REVIEW

File: src/Stratis.Features.FederatedPeg/Wallet/FederationWalletSyncManager.cs [hunk 1]
```diff
@@ -208,6 +223,36 @@ public virtual void ProcessBlock(Block block)
             this.walletManager.ProcessBlock(block, newTip);
         }
 
+        /// <inheritdoc />
+        public virtual void ProcessBlock(Block block)
+        {
+            Guard.NotNull(block, nameof(block));
+
+            // If the queue reaches the maximum limit, ignore incoming blocks until the queue is empty.
+            if (!this.maxQueueSizeReached)
```
Comment (quantumagi) [inline src/Stratis.Features.FederatedPeg/Wallet/FederationWalletSyncManager.cs]: Can we really simply ignore blocks?

**Comment by codingupastorm** [comment]
@quantumagi could you elaborate a bit more? The main point of this is to avoid going into Thread.Sleeps during consensus updating in IBD so I've just gone for the simplest approach which is to borrow the queue approach from the standard wallet.

What determinism issues do you see could arise? I don't see how this approach could be less predictable than what we had before

File: src/Stratis.Features.FederatedPeg/Wallet/FederationWalletSyncManager.cs (hunk 1 above)
Comment (codingupastorm) [reply src/Stratis.Features.FederatedPeg/Wallet/FederationWalletSyncManager.cs]: This is only going to happen in IBD, and what will happen is after the next batch of blocks comes in, the `OnProcessBlockAsync` method will retrieve any that are missing beforehand and continue where it left off.

**Comment by quantumagi** [comment]
A method calling `ProcessBlock` may have certain expectation as to the wallet state afterwards. If you can confirm that there are no such calls, i.e. calls having an expectation of an updated wallet directly following the call, then its not an issue. 

That said, it may still be a good idea to work towards allowing blocks to be passed in batches throughout the code base where calls may lead to persistence. Queuing does not reduce work load or provide efficiency when the bottleneck is the db, only batching provides gains.

**Comment by codingupastorm** [comment]
Okay, I hear what you're saying. Just to clarify, this is is an issue with both the current wallet in the full node, and the federation wallet as it currently exists. 

I'm not going to fix the way that the wallets sync blocks now, but it's worth looking at in the future. 

For now, this will enable for IBD to be significantly sped up, and prevents Thread.Sleep happening synchronously with all of the other block notification subscribers

**Pull request merged**